utoipa-gen/src/openapi.rs

```diff
@@ -269,45 +288,48 @@ impl ToTokens for OpenApi {
 }
 
 fn impl_components(
-    components: &Punctuated<Component, Comma>,
-    tokens: &mut TokenStream,
+    schemas: &Punctuated<Component, Comma>,
+    responses: &Punctuated<Responses, Comma>,
 ) -> Option<TokenStream> {
-    if !components.is_empty() {
-        let mut components_tokens = components.iter().fold(
+    if !(schemas.is_empty() && responses.is_empty()) {
+        let builder_tokens = schemas.iter().fold(
             quote! { utoipa::openapi::ComponentsBuilder::new() },
-            |mut schema, component| {
+            |mut builder_tokens, component| {
                 let path = &component.path;
                 let ident = component.get_ident().unwrap();
                 let span = ident.span();
                 let component_name = &*ident.to_string();
 
                 let (_, ty_generics, _) = component.generics.split_for_impl();
 
-                let assert_ty_generics = if component.has_lifetime_generics() {
-                    Some(quote! {<'static>})
-                } else {
-                    Some(ty_generics.to_token_stream())
-                };
-                let assert_component = format_ident!("_AssertComponent{}", component_name);
-                tokens.extend(quote_spanned! {span=>
-                    struct #assert_component where #path #assert_ty_generics: utoipa::Component;
-                });
-
```

Comment:
Seems like the assert component is being removed here? I had previously added these assert components for better error reports when someone is missing the `Component` derive from the item.

Reply:
Sorry I should have mentioned this in the description for the PR. I found that the `<#component as utoipa::Component>::component()` in a spanned quote, the message is pretty clear, there doesn't appear to be any benefit of having the assert, or maybe I'm missing something?

 If I remove the `Component` derive for `Todo` in the `todo-warp` example project:

With the assert:

```text
error[E0277]: the trait bound `Todo: utoipa::Component` is not satisfied
  --> src/main.rs:24:20
   |
24 |         components(todo::Todo),
   |                    ^^^^^^^^^^ the trait `utoipa::Component` is not implemented for `Todo`
   |
   = help: see issue #48214

For more information about this error, try `rustc --explain E0277`.
```

Without assert, in this PR:

```text
error[E0277]: the trait bound `Todo: utoipa::Component` is not satisfied
  --> src/main.rs:24:26
   |
24 |         components(todo::Todo),
   |                          ^^^^ the trait `utoipa::Component` is not implemented for `Todo`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `todo-warp` due to previous error
```

Reply:
Actually thats great. :slightly_smiling_face: Then we can remove these assert occurences there completely. There are few other's as well too.